cylc/flow/config_schema.py

```diff
@@ -0,0 +1,660 @@
+            'inactivity handler': [VDR.V_STRING_LIST, None],
+            'shutdown handler': [VDR.V_STRING_LIST, None],
+            'aborted handler': [VDR.V_STRING_LIST, None],
+            'stalled handler': [VDR.V_STRING_LIST, None],
+            'timeout': [VDR.V_INTERVAL],
+            'inactivity': [VDR.V_INTERVAL],
+            'abort if startup handler fails': [VDR.V_BOOLEAN],
+            'abort if shutdown handler fails': [VDR.V_BOOLEAN],
+            'abort if timeout handler fails': [VDR.V_BOOLEAN],
+            'abort if inactivity handler fails': [VDR.V_BOOLEAN],
+            'abort if stalled handler fails': [VDR.V_BOOLEAN],
+            'abort on stalled': [VDR.V_BOOLEAN],
+            'abort on timeout': [VDR.V_BOOLEAN],
+            'abort on inactivity': [VDR.V_BOOLEAN],
+            'mail events': [VDR.V_STRING_LIST, None],
+            'mail from': [VDR.V_STRING],
+            'mail smtp': [VDR.V_STRING],
+            'mail to': [VDR.V_STRING],
+            'mail footer': [VDR.V_STRING],
+        },
+        'reference test': {
+            'suite shutdown event handler': [
+                VDR.V_STRING, 'cylc hook check-triggering'],
+            'required run mode': [
+                VDR.V_STRING,
+                '', 'live', 'simulation', 'dummy-local', 'dummy'],
+            'allow task failures': [VDR.V_BOOLEAN],
+            'expected task failures': [VDR.V_STRING_LIST],
+            'live mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+            'dummy mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+            'dummy-local mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+            'simulation mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+        },
+        'authorization': {
+            # Allow owners to grant public shutdown rights at the most, not
+            # full control.
+            'public': (
+                [VDR.V_STRING, '']
+                + [level.name.lower().replace('_', '-') for level in [
+                    Priv.IDENTITY, Priv.DESCRIPTION, Priv.STATE_TOTALS,
+                    Priv.READ, Priv.SHUTDOWN]])
+        },
+    },
+    'document viewers': {
+        'html': [VDR.V_STRING, 'firefox'],
+    },
+    'documentation': {
+        'local': [VDR.V_STRING, ''],
+        'online': [VDR.V_STRING,
+                   'http://cylc.github.io/doc/built-sphinx/index.html'],
+        'cylc homepage': [VDR.V_STRING, 'http://cylc.github.io/'],
+    },
+    'editors': {
+        'terminal': [VDR.V_STRING, 'vim'],
+        'gui': [VDR.V_STRING, 'gvim -f'],
+    },
+    'job platforms': {
+        'default platform': {
+            'run directory': [VDR.V_STRING, '$HOME/cylc-run'],
+            'work directory': [VDR.V_STRING, '$HOME/cylc-run'],
+            'task communication method': [
+                VDR.V_STRING, 'default', 'ssh', 'poll'],
```

Comment:
To avoid complexity and for clarity, we should probably use `zmq`, `ssh+zmq`, and `poll` (or `none`).